### ISSUE TITLE
`module.exports` does not always act like `exports`.

### ISSUE THREAD

**Issue by mkantor** [jsdoc3/jsdoc#904]
Based on JSDoc's documentation (and common sense) I would expect things assigned to `module.exports` and `exports` to act the same, but this is not the case.

The following code produces the expected output, with `require("path/to/module")` documented as a class which has a method named `foo()`.
```javascript
/**
 * Example of assigning a constructor to plain old exports and methods to
 * its prototype.
 * @module
 */

/** @constructor */
module.exports = exports = function() {};

/** An instance method. */
exports.prototype.foo = function() {};
```

---

However, this version does *not* document the class, and `foo()` is nowhere to be seen in the generated docs.
```javascript
/**
 * Example of assigning a constructor to module.exports and methods to its
 * prototype.
 * @module
 */

/** @constructor */
module.exports = function() {};

/** An instance method. */
module.exports.prototype.foo = function() {};
```

**Comment by hegemonic** [comment]
The second example works just fine for me--the generated docs show the class and the `foo()` instance method. What version of JSDoc are you using?

**Comment by mkantor** [comment]
I'm running 3.2.2, which as far as I can tell is the latest stable release.

In case I'm doing something wrong, what follows should be an easily-reproducible example using [this gist](https://gist.github.com/mkantor/255d22f9c3fa918387dd).

```bash
$ git clone https://gist.github.com/255d22f9c3fa918387dd.git
Cloning into '255d22f9c3fa918387dd'...
remote: Counting objects: 3, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 3 (delta 0), reused 0 (delta 0)
Unpacking objects: 100% (3/3), done.
Checking connectivity... done.
```
```bash
$ jsdoc 255d22f9c3fa918387dd/jsdoc-incorrect-example.js 
```
```bash
$ cat out/module-jsdoc-incorrect-example.html 
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Module: jsdoc-incorrect-example</title>
    
    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">
    
    <h1 class="page-title">Module: jsdoc-incorrect-example</h1>
    
    



<section>
    
<header>
    <h2>
    jsdoc-incorrect-example
    </h2>
    
</header>  

<article>
    <div class="container-overview">
    
        
<dt>
    <h4 class="name" id="module:jsdoc-incorrect-example"><span class="type-signature"></span>new require("jsdoc-incorrect-example")<span class="signature">()</span><span class="type-signature"></span></h4>
    
    
</dt>
<dd>
    
    

    
    
    
    
    
    
    
<dl class="details">
    
        
    
    
    

    
    
    
    
    
    
    
    
    
    
    
    
    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="jsdoc-incorrect-example.js.html">jsdoc-incorrect-example.js</a>, <a href="jsdoc-incorrect-example.js.html#line8">line 8</a>
    </li></ul></dd>
    
    
    
    
    
    
    
</dl>

    
    

    

    
    
    
    
    
    
    
</dd>

    
    
    
        
            <div class="description">Example of assigning a constructor to module.exports and methods to its
prototype.</div>
        
        
        
<dl class="details">
    
        
    
    
    

    
    
    
    
    
    
    
    
    
    
    
    
    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="jsdoc-incorrect-example.js.html">jsdoc-incorrect-example.js</a>, <a href="jsdoc-incorrect-example.js.html#line1">line 1</a>
    </li></ul></dd>
    
    
    
    
    
    
    
</dl>

        
        
    
    </div>
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
</article>

</section>  




</div>

<nav>
    <h2><a href="index.html">Index</a></h2><h3>Modules</h3><ul><li><a href="module-jsdoc-incorrect-example.html">jsdoc-incorrect-example</a></li></ul>
</nav>

<br clear="both">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.2.2</a> on Thu Feb 05 2015 21:09:55 GMT-0800 (PST)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>
```
```bash
$ jsdoc --version
JSDoc 3.2.2 (Mon, 04 Nov 2013 16:12:50 GMT)
```

It's ugly, but you'll notice that the HTML does not contain the correct class/method documentation.

**Comment by hegemonic** [comment]
@mkantor Thanks for clarifying. I believe that this issue is fixed in JSDoc 3.3.0, which should be released within a week or so.
